internal/delete/connector/execute.go

```diff
@@ -14,23 +14,86 @@
+	"fmt"
+
+	"github.com/eclipse-iofog/iofogctl/internal"
 	"github.com/eclipse-iofog/iofogctl/internal/config"
+	"github.com/eclipse-iofog/iofogctl/internal/execute"
 	"github.com/eclipse-iofog/iofogctl/pkg/util"
 )
 
-func Execute(namespace, name string) error {
-	util.SpinStart("Deleting Connector")
+type executor struct {
+	name        string
+	namespace   string
+	useDetached bool
+	soft        bool
+}
+
+func NewExecutor(namespace, name string, useDetached, soft bool) (execute.Executor, error) {
+	return executor{name: name, namespace: namespace, useDetached: useDetached, soft: soft}, nil
+}
 
-	// Get executor
-	exe, err := NewExecutor(namespace, name)
-	if err != nil {
-		return err
+func (exe executor) GetName() string {
+	return exe.name
+}
+
+func (exe executor) Execute() error {
+	if !exe.useDetached {
+		// Try to delete connector from Controller database
+		ctrlClient, err := internal.NewControllerClient(exe.namespace)
+		if err == nil {
+			connectors, err := ctrlClient.ListConnectors()
+			if err != nil {
+				util.PrintInfo(fmt.Sprintf("Could not delete connector %s from the Controller. Error: %s\n", exe.name, err.Error()))
+			} else {
+				for _, connector := range connectors.Connectors {
+					if connector.Name == exe.name {
+						if err = ctrlClient.DeleteConnector(connector.IP); err != nil {
+							util.PrintInfo(fmt.Sprintf("Could not delete connector %s from the Controller. Error: %s\n", exe.name, err.Error()))
+						}
+					}
+				}
+			}
+		}
 	}
 
-	// Execute deletion
-	if err = exe.Execute(); err != nil {
-		return err
+	// Try to remove iofog-connector
+	var connector config.Connector
+	var err error
+	if exe.useDetached {
+		connector, err = config.GetDetachedConnector(exe.name)
+	} else {
+		connector, err = config.GetConnector(exe.namespace, exe.name)
 	}
+	if err == nil {
+		// Don't delete stack if soft delete
+		if !exe.soft {
+			if util.IsLocalHost(connector.Host) {
+				if err = exe.localRemove(); err != nil {
+					util.PrintInfo(fmt.Sprintf("Could not remove iofog-connector container. Error: %s\n", err.Error()))
+				}
+			} else if connector.Kube.Config != "" {
+				if err = exe.k8sRemove(); err != nil {
+					util.PrintInfo(fmt.Sprintf("Could not remove iofog-connector from Kubernetes. Error: %s\n", err.Error()))
+				}
+			} else {
+				if err = exe.remoteRemove(); err != nil {
+					util.PrintInfo(fmt.Sprintf("Could not remove iofog-connector from the host %s. Error: %s\n", connector.Host, err.Error()))
+				}
+			}
+		}
 
-	return config.Flush()
+		// Update config
+		if exe.useDetached {
+			return config.DeleteDetachedConnector(exe.name)
+		}
+		if err = config.DeleteConnector(exe.namespace, exe.name); err != nil {
+			util.PrintInfo(fmt.Sprintf("Could not remove Connector from iofogctl config. Error: %s\n", err.Error()))
+		} else {
+			defer config.Flush()
+		}
+	} else {
+		util.PrintInfo(fmt.Sprintf("Could not find iofog-connector in iofogctl configuration. Error: %s\n", err.Error()))
```

Comment:
why error not fatal?

Reply:
> why error not fatal?

Same as for agent, if you have a corrupted config file, you would still want to remove the connector from the Controller database